packages/table/preview/perf.tsx

```diff
@@ -183,9 +363,101 @@ class MutableTable extends React.Component<{}, IMutableTableState> {
         return this.store.set(rowIndex, columnIndex, value);
     }
+
+    private updateBooleanState(stateKey: keyof IMutableTableState) {
+        return handleBooleanChange((value: boolean) => {
+            this.setState({ [stateKey]: value });
+        });
+    }
+
+    private updateNumberState(stateKey: keyof IMutableTableState) {
+        return handleNumberChange((value: number) => {
+            this.setState({ [stateKey]: value });
+        });
+    }
+
+    private renderBodyContextMenu = () => {
+        const menu = (
+            <Menu>
+                <MenuItem iconName="search-around" text="Item 1" />
+                <MenuItem iconName="search" text="Item 2" />
+                <MenuItem iconName="graph-remove" text="Item 3" />
+                <MenuItem iconName="group-objects" text="Item 4" />
+                <MenuDivider />
+                <MenuItem disabled={true} text="Disabled item" />
+            </Menu>
+        );
+        return this.state.enableContextMenu ? menu : undefined;
+    }
+
+    private getEnabledSelectionModes() {
+        const selectionModes: RegionCardinality[] = [];
+        if (this.state.enableFullTableSelection) {
+            selectionModes.push(RegionCardinality.FULL_TABLE);
+        }
+        if (this.state.enableColumnSelection) {
+            selectionModes.push(RegionCardinality.FULL_COLUMNS);
+        }
+        if (this.state.enableRowSelection) {
+            selectionModes.push(RegionCardinality.FULL_ROWS);
+        }
+        if (this.state.enableCellSelection) {
+            selectionModes.push(RegionCardinality.CELLS);
+        }
+        return selectionModes;
+    }
+
+    private getEnabledLoadingOptions() {
+        const loadingOptions: TableLoadingOption[] = [];
+        if (this.state.showColumnHeadersLoading) {
+            loadingOptions.push(TableLoadingOption.COLUMN_HEADERS);
+        }
+        if (this.state.showRowHeadersLoading) {
+            loadingOptions.push(TableLoadingOption.ROW_HEADERS);
+        }
+        if (this.state.showCellsLoading) {
+            loadingOptions.push(TableLoadingOption.CELLS);
+        }
+        return loadingOptions;
+    }
+
+    private getStyledRegionGroups() {
+        // show 3 styled regions as samples
+        return !this.state.showCustomRegions ? [] : [
+            {
+                className: "tbl-styled-region-success",
+                regions: [Regions.cell(0, 0, 3, 3)],
+            },
+            {
+                className: "tbl-styled-region-warning",
+                regions: [Regions.cell(2, 1, 8, 1)],
+            },
+            {
+                className: "tbl-styled-region-danger",
+                regions: [Regions.cell(5, 3, 7, 7)],
+            },
+        ] as IStyledRegionGroup[];
+    }
 }
 
 ReactDOM.render(
     <MutableTable/>,
-    document.querySelector(".table"),
+    document.querySelector("#page-content"),
 );
+
+// TODO: Pull these from @blueprintjs/docs
```

Comment:
Happy to do this in this PR. Wasn't able to wrangle it after a few seconds of poking, so I copied+pasted to unblock myself.